src/clj/game/core/runs.clj

```diff
@@ -648,14 +652,17 @@
                                   (next-access state side eid already-accessed card)
                                   (effect-completed state side eid))))))}))
 
-(defmethod choose-access :archives [cards server]
+(defmethod choose-access :archives [cards server {:keys [no-root] :as args}]
   {:async true
    :effect (req (let [cards (concat (get-archives-accessible state) (-> @state :corp :servers :archives :content))
                       archives-count (+ (count (-> @state :corp :discard)) (count (-> @state :corp :servers :archives :content)))]
                   (if (not-empty cards)
                     (if (= 1 archives-count)
-                      (access-card state side eid (first cards))
-                      (continue-ability state side (access-helper-archives state archives-count #{}) card nil))
+                      (wait-for (access-card state side eid (first cards)))
```

Comment:
If you're trying to make the line below wait for `access-card` to finish, you have to encompass the following lines in the `wait-for`'s parenthesis.

Reply:
You also don't pass your own `eid` to another function that you are awaiting, so remove that parameter.

Reply:
I see now that the `continue-ability` is for the false if-branch, but removing the `wait-for` breaks things, so I don't actually know what's going on here.